packages/truffle-core/lib/commands/migrate.js

```diff
@@ -129,6 +129,42 @@ const command = {
     return { dryRunOnly, dryRunAndMigrations };
   },
 
+  prepareConfigForRealMigrations: async function(buildDir, options, done) {
+    const Artifactor = require("truffle-artifactor");
+    const Resolver = require("truffle-resolver");
+    const Migrate = require("truffle-migrate");
+    const Config = require("truffle-config");
+    const Environment = require("../environment");
+
+    let accept = true;
+
+    if (options.interactive) {
+      accept = await Migrate.acceptDryRun();
+    }
+
+    if (accept) {
+      const config = Config.detect(options);
+
+      config.contracts_build_directory = buildDir;
+      config.artifactor = new Artifactor(buildDir);
+      config.resolver = new Resolver(config);
+
+      try {
+        await Environment.detect(config);
+      } catch (error) {
+        throw new Error(error);
```

Comment:
By throwing this error here, we won't end up running `done(error)`. Not sure if that's intended? My hunch is that this line needs to be removed.

Reply:
Oh good catch. Actually `done` gets called on the outside of the function...`done` should be removed from the function signature. I think this was some code that got left in along with the callback to `done` in the else.

Reply:
Ok so prepareConfigForRealMigrations is only called in one place and now `done` is handled on the outside of the function.